frontend/src/pages/Blogs.tsx

```diff
@@ -15,33 +15,68 @@ export interface BlogType {
   claps: [];
   bookmarkId?: string;
 }
-
 const Blogs = () => {
   const { blogs, loading } = useBlogs();
 
+  const [searchQuery, setSearchQuery] = useState("");
+  const [searchedData, setSearchedData] = useState([]);
+
+  // Search-Logic
+  useEffect(() => {
+    if (searchQuery) {
+      const debouncedSearch = setTimeout(() => {
+        setSearchedData(
+          blogs.filter((info: any) => {
```

Comment:
This is not a correct approach. 
We shouldn't be filtering on `blogs`.

The debounced query should go to the backend as an API request, and the backend should return a list of filtered blogs.

Please change the approach.

Reply:
Okay so instead of filtering the data received from the backend by /blogs route. Do it directly by fetching the data from the backend when anything is being searched.

Reply:
Correct. You will need to make backend changes for this since it's not implemented in any api yet.

Reply:
You can checkout the `blogRouter.get("/bulk/:id", async(c)=>...)` route for reference.